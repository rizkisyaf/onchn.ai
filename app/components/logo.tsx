import { motion } from 'framer-motion'
import Image from 'next/image'

interface LogoProps {
  size?: number
  className?: string
}

export function Logo({ size = 40, className = '' }: LogoProps) {
  return (
    <motion.div 
      className={`relative ${className}`}
      style={{ width: size, height: size }}
      whileHover={{ scale: 1.1 }}
      whileTap={{ scale: 0.95 }}
    >
      <Image
        src="/images/logo/logo-mark.svg"
        alt="onchn.ai logo"
        width={size}
        height={size}
        className="w-full h-full"
      />
    </motion.div>
  )
} 