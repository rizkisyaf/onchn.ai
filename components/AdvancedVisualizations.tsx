import React from 'react';
import { ResponsiveHeatMap } from '@nivo/heatmap';
import { ResponsiveNetwork } from '@nivo/network';
import { WalletData, Transaction } from '@/types/wallet';
import { DefaultHeatMapDatum } from '@nivo/heatmap';
import { HeatMapSerie } from '@nivo/heatmap';

interface AdvancedVisualizationsProps {
  walletData: WalletData[];
  transactions: Transaction[];
}

interface HeatMapData {
  id: string;
  data: DefaultHeatMapDatum[];
}

interface NetworkNode {
  id: string;
  size: number;
  color: string;
}

interface NetworkLink {
  source: string;
  target: string;
  value: number;
  distance: number;
  thickness?: number;
  color?: string;
}

interface NetworkData {
  nodes: NetworkNode[];
  links: NetworkLink[];
}

export const AdvancedVisualizations: React.FC<AdvancedVisualizationsProps> = ({ walletData, transactions }) => {
  const heatmapData: HeatMapSerie<DefaultHeatMapDatum, { wallet: string }>[] = walletData.map((item) => ({
    id: item.wallet,
    wallet: item.wallet,
    data: [{ x: 'value', y: item.value }],
  }));

  const networkData: NetworkData = {
    nodes: walletData.map((item) => ({
      id: item.wallet,
      size: item.value,
      color: `hsl(${Math.random() * 360}, 70%, 50%)`,
    })),
    links: transactions.map((tx) => ({
      source: tx.source,
      target: tx.target,
      value: tx.value,
      distance: 100,
    })),
  };

  return (
    <div className="grid grid-cols-1 md:grid-cols-2 gap-8">
      <div className="h-[400px]">
        <ResponsiveHeatMap
          data={heatmapData}
          margin={{ top: 60, right: 90, bottom: 60, left: 90 }}
          valueFormat=">-.2s"
          axisTop={{
            tickSize: 5,
            tickPadding: 5,
            tickRotation: -90,
            legend: '',
            legendOffset: 46,
          }}
          axisRight={null}
          axisBottom={{
            tickSize: 5,
            tickPadding: 5,
            tickRotation: -90,
            legend: '',
            legendOffset: 46,
          }}
          axisLeft={{
            tickSize: 5,
            tickPadding: 5,
            tickRotation: 0,
            legend: 'Wallets',
            legendPosition: 'middle',
            legendOffset: -72,
          }}
          colors={{
            type: 'sequential',
            scheme: 'blues',
          }}
          emptyColor="#555555"
          legends={[
            {
              anchor: 'bottom',
              translateX: 0,
              translateY: 30,
              length: 400,
              thickness: 8,
              direction: 'row',
              tickPosition: 'after',
              tickSize: 3,
              tickSpacing: 4,
              tickOverlap: false,
              tickFormat: '>-.2s',
              title: 'Value â†’',
              titleAlign: 'start',
              titleOffset: 4,
            },
          ]}
        />
      </div>
      <div className="h-[400px]">
        <ResponsiveNetwork
          data={networkData}
          margin={{ top: 0, right: 0, bottom: 0, left: 0 }}
          linkDistance={(e) => e.distance}
          centeringStrength={0.3}
          repulsivity={6}
          nodeSize={(n) => n.size}
          nodeColor={(e) => e.color}
          linkThickness={(n) => 2 + 2 * (n as unknown as NetworkLink).value}
          activeNodeSize={(n) => 1.5 * n.size}
          inactiveNodeSize={(n) => n.size}
          nodeTooltip={({ node }) => (
            <div className="bg-white p-2 rounded shadow">
              <strong>{node.id}</strong>
              <br />
              Value: {node.size}
            </div>
          )}
        />
      </div>
    </div>
  );
};

