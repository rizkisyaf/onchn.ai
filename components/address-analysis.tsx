'use client'

import { useEffect } from 'react'
import { Card } from '@/components/ui/card'
import { Progress } from '@/components/ui/progress'
import { useAddressDetection } from '@/hooks/use-address-detection'
import { AddressMetadata } from '@/types/wallet'

interface AddressAnalysisProps {
  address: string
  onDetection?: (metadata: AddressMetadata) => void
}

export function AddressAnalysis({ address, onDetection }: AddressAnalysisProps) {
  const { detect, isDetecting, error } = useAddressDetection({
    onDetection,
  })

  useEffect(() => {
    if (address) {
      detect(address)
    }
  }, [address, detect])

  if (error) {
    return (
      <Card className="p-4">
        <div className="text-red-500">
          <h3 className="font-semibold">Error</h3>
          <p>{error.message}</p>
        </div>
      </Card>
    )
  }

  if (isDetecting) {
    return (
      <Card className="p-4">
        <div className="space-y-4">
          <h3 className="font-semibold">Detecting Address Type...</h3>
          <Progress value={50} />
        </div>
      </Card>
    )
  }

  return (
    <Card className="p-4">
      <div className="space-y-4">
        <h3 className="font-semibold">Address Analysis</h3>
        <p className="text-sm text-muted-foreground">
          Select an address to analyze its type and properties.
        </p>
      </div>
    </Card>
  )
} 