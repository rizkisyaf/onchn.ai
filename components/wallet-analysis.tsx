'use client'

import { useEffect, useState } from 'react'
import { motion } from 'framer-motion'
import { Card } from '@/components/ui/card'
import { Progress } from '@/components/ui/progress'
import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
import { ArrowUpDown, Activity, Shield, Network } from 'lucide-react'

interface WalletAnalysisProps {
  address: string
}

export function WalletAnalysis({ address }: WalletAnalysisProps) {
  const [riskScore, setRiskScore] = useState(75)
  const [transactions, setTransactions] = useState([
    { type: 'send', amount: '10.5', token: 'SOL', time: '2 mins ago' },
    { type: 'receive', amount: '25.0', token: 'USDC', time: '5 mins ago' },
    { type: 'swap', amount: '5.2', token: 'SOL', time: '10 mins ago' },
  ])

  return (
    <div className="space-y-6">
      {/* Risk Score */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
      >
        <div className="bg-white/5 rounded-xl p-6 backdrop-blur-sm border border-white/10">
          <div className="flex items-center justify-between mb-4">
            <div>
              <h3 className="text-xl font-semibold">Risk Score</h3>
              <p className="text-gray-400">Overall wallet safety assessment</p>
            </div>
            <div className="text-right">
              <p className="text-3xl font-bold bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent">
                {riskScore}%
              </p>
              <p className="text-sm text-gray-400">Safe</p>
            </div>
          </div>
          <Progress value={riskScore} className="h-2" />
        </div>
      </motion.div>

      {/* Analysis Tabs */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.1 }}
      >
        <Tabs defaultValue="transactions" className="space-y-4">
          <TabsList className="bg-white/5 border-b border-white/10">
            <TabsTrigger value="transactions" className="data-[state=active]:text-primary">
              Transactions
            </TabsTrigger>
            <TabsTrigger value="activity" className="data-[state=active]:text-primary">
              Activity
            </TabsTrigger>
            <TabsTrigger value="connections" className="data-[state=active]:text-primary">
              Connections
            </TabsTrigger>
          </TabsList>

          <TabsContent value="transactions" className="space-y-4">
            {transactions.map((tx, i) => (
              <div key={i} className="bg-white/5 rounded-xl p-4 backdrop-blur-sm border border-white/10 flex items-center justify-between">
                <div className="flex items-center gap-4">
                  <ArrowUpDown className="text-primary" />
                  <div>
                    <p className="font-medium">{tx.type.charAt(0).toUpperCase() + tx.type.slice(1)}</p>
                    <p className="text-sm text-gray-400">{tx.time}</p>
                  </div>
                </div>
                <p className="font-medium">
                  {tx.type === 'receive' ? '+' : '-'}{tx.amount} {tx.token}
                </p>
              </div>
            ))}
          </TabsContent>

          <TabsContent value="activity" className="space-y-4">
            <div className="grid grid-cols-1 md:grid-cols-3 gap-6">
              <div className="bg-white/5 rounded-xl p-6 backdrop-blur-sm border border-white/10">
                <Activity className="h-8 w-8 text-primary mb-2" />
                <h4 className="font-medium mb-1">Transaction Volume</h4>
                <p className="text-2xl font-bold bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent">$25,420</p>
                <p className="text-sm text-gray-400">Last 30 days</p>
              </div>
              <div className="bg-white/5 rounded-xl p-6 backdrop-blur-sm border border-white/10">
                <Shield className="h-8 w-8 text-primary mb-2" />
                <h4 className="font-medium mb-1">Security Score</h4>
                <p className="text-2xl font-bold bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent">92/100</p>
                <p className="text-sm text-gray-400">Very Good</p>
              </div>
              <div className="bg-white/5 rounded-xl p-6 backdrop-blur-sm border border-white/10">
                <Network className="h-8 w-8 text-primary mb-2" />
                <h4 className="font-medium mb-1">Network Usage</h4>
                <p className="text-2xl font-bold bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent">156</p>
                <p className="text-sm text-gray-400">Connected dApps</p>
              </div>
            </div>
          </TabsContent>

          <TabsContent value="connections" className="space-y-4">
            <div className="bg-white/5 rounded-xl p-6 backdrop-blur-sm border border-white/10">
              <h4 className="font-medium mb-4">Connected Wallets</h4>
              <div className="space-y-4">
                {[1, 2, 3].map((_, i) => (
                  <div key={i} className="flex items-center justify-between p-4 bg-white/5 rounded-xl">
                    <div>
                      <p className="font-medium">Wallet {i + 1}</p>
                      <p className="text-sm text-gray-400">7X...4z</p>
                    </div>
                    <p className="text-sm text-gray-400">15 transactions</p>
                  </div>
                ))}
              </div>
            </div>
          </TabsContent>
        </Tabs>
      </motion.div>
    </div>
  )
}

