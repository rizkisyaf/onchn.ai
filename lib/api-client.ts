import { z } from 'zod'

export interface APIError extends Error {
  code: 'RATE_LIMIT' | 'NOT_FOUND' | 'UNAUTHORIZED' | 'INTERNAL_ERROR'
  message: string
  details?: any
}

export class APIErrorImpl extends Error implements APIError {
  constructor(
    public code: APIError['code'],
    message: string,
    public details?: any
  ) {
    super(message)
    this.name = 'APIError'
  }
}

export const WalletDataSchema = z.object({
  address: z.string(),
  wallet: z.string(),
  value: z.number(),
  balance: z.number(),
  tokens: z.array(z.object({
    address: z.string(),
    symbol: z.string(),
    name: z.string(),
    decimals: z.number(),
    balance: z.number(),
    value: z.number(),
    price: z.number(),
    change24h: z.number()
  })),
  transactions: z.array(z.object({
    hash: z.string(),
    timestamp: z.number(),
    type: z.enum(['swap', 'transfer', 'stake', 'unstake', 'other']),
    amount: z.number(),
    value: z.number(),
    source: z.string(),
    target: z.string(),
    from: z.string(),
    to: z.string(),
    fee: z.number(),
    success: z.boolean(),
    programId: z.string().optional(),
    data: z.any().optional()
  })),
  stats: z.object({
    totalValue: z.number(),
    totalTransactions: z.number(),
    uniqueTokens: z.number(),
    avgTransactionValue: z.number(),
    lastActivity: z.number()
  }),
  avgTransactionValue: z.number(),
  totalInflow: z.number(),
  totalOutflow: z.number(),
  uniqueInteractions: z.number(),
  dcaFrequency: z.number(),
  avgDcaAmount: z.number(),
  winRate: z.number(),
  pnl: z.number()
})

export type WalletData = z.infer<typeof WalletDataSchema>

export async function getWalletData(walletAddress: string): Promise<WalletData> {
  try {
    const response = await fetch(`${process.env.SOLANATRACKER_API_URL}/wallet/${walletAddress}`, {
      headers: {
        'Content-Type': 'application/json',
        'x-api-key': process.env.SOLANATRACKER_API_KEY || '',
      },
    })

    if (!response.ok) {
      throw new APIErrorImpl('INTERNAL_ERROR', `Failed to fetch wallet data: ${response.statusText}`)
    }

    const data = await response.json()
    return WalletDataSchema.parse(data)
  } catch (error) {
    if (error instanceof z.ZodError) {
      throw new APIErrorImpl('INTERNAL_ERROR', 'Invalid wallet data format')
    }
    throw error
  }
}

export async function getWalletInfo(walletAddress: string): Promise<WalletData> {
  return getWalletData(walletAddress)
}

export async function getWalletTransactions(walletAddress: string): Promise<any[]> {
  try {
    const response = await fetch(`${process.env.SOLANATRACKER_API_URL}/wallet/${walletAddress}/transactions`, {
      headers: {
        'Content-Type': 'application/json',
        'x-api-key': process.env.SOLANATRACKER_API_KEY || '',
      },
    })

    if (!response.ok) {
      throw new APIErrorImpl('INTERNAL_ERROR', `Failed to fetch wallet transactions: ${response.statusText}`)
    }

    const data = await response.json()
    return data
  } catch (error) {
    throw error
  }
}

