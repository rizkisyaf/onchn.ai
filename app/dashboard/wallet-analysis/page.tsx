'use client'

import { useState } from 'react'
import { motion } from 'framer-motion'
import { WalletAnalysis } from '@/components/wallet-analysis'
import { Card } from '@/components/ui/card'
import { Input } from '@/components/ui/input'
import { Button } from '@/components/ui/button'
import { useToast } from '@/components/ui/use-toast'
import { Search as SearchIcon, Loader2, Wallet as WalletIcon } from 'lucide-react'
import { ErrorBoundary } from '@/components/error-boundary'
import { useSubscription } from '@/hooks/useSubscription'

export default function WalletAnalysisPage() {
  const { canAccessFeature, currentPlan } = useSubscription()
  const canAccessWalletAnalysis = canAccessFeature('advancedAnalytics')
  const [address, setAddress] = useState('')
  const [isAnalyzing, setIsAnalyzing] = useState(false)
  const { toast } = useToast()

  if (!canAccessWalletAnalysis) {
    return (
      <div className="p-6">
        <Card className="p-6 bg-white/5 border-white/10">
          <h2 className="text-xl font-semibold mb-4">Feature Restricted</h2>
          <p className="text-gray-400 mb-4">
            Advanced wallet analysis is only available for paid plans. Upgrade your subscription to access this feature.
          </p>
          <Button>Upgrade Now</Button>
        </Card>
      </div>
    )
  }

  const handleAnalyze = () => {
    if (!address) {
      toast({
        title: 'Error',
        description: 'Please enter a wallet address'
      })
      return
    }

    setIsAnalyzing(true)
    // Add error handling for analysis
    try {
      // Analysis logic here
    } catch (error) {
      toast({
        title: 'Error',
        description: 'Failed to analyze wallet. Please try again.'
      })
    } finally {
      setIsAnalyzing(false)
    }
  }

  return (
    <div className="space-y-8">
      {/* Header */}
      <div className="flex items-center gap-3">
        <WalletIcon className="h-8 w-8 text-primary" />
        <motion.h1 
          initial={{ opacity: 0, y: 20 }}
          animate={{ opacity: 1, y: 0 }}
          className="text-3xl font-bold"
        >
          Wallet Analysis
        </motion.h1>
      </div>

      {/* Search */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.1 }}
      >
        <Card className="p-6 bg-white/5 border-white/10">
          <div className="flex flex-col md:flex-row gap-4">
            <Input
              type="text"
              placeholder="Enter wallet address"
              value={address}
              onChange={(e) => setAddress(e.target.value)}
              className="flex-1 bg-white/5 border-white/10"
            />
            <Button 
              onClick={handleAnalyze}
              disabled={isAnalyzing}
              className="bg-primary hover:bg-primary/90"
            >
              {isAnalyzing ? (
                <>
                  <Loader2 className="mr-2 h-4 w-4 animate-spin" />
                  Analyzing...
                </>
              ) : (
                <>
                  <SearchIcon className="mr-2 h-4 w-4" />
                  Analyze
                </>
              )}
            </Button>
          </div>
        </Card>
      </motion.div>

      {/* Analysis Results */}
      <motion.div
        initial={{ opacity: 0, y: 20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ delay: 0.2 }}
      >
        <WalletAnalysis address={address} />
      </motion.div>
    </div>
  )
} 