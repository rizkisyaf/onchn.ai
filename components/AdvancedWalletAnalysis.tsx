import React from 'react'
import { Card } from '@/components/ui'
import { WalletData } from '@/types/wallet'
import { LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer, PieChart, Pie, Cell } from 'recharts'

interface AdvancedWalletAnalysisProps {
  walletData: WalletData
}

const COLORS = ['#0088FE', '#00C49F', '#FFBB28', '#FF8042', '#8884D8']

export const AdvancedWalletAnalysis: React.FC<AdvancedWalletAnalysisProps> = ({ walletData }) => {
  const tokenDistribution = walletData.tokens.map(token => ({
    name: token.symbol,
    value: token.value
  }))

  const totalValue = walletData.tokens.reduce((sum, token) => sum + token.value, 0)

  const calculateDiversificationScore = () => {
    const herfindahlIndex = walletData.tokens.reduce((sum, token) => {
      const marketShare = token.value / totalValue
      return sum + marketShare * marketShare
    }, 0)
    return (1 - herfindahlIndex) * 100
  }

  const diversificationScore = calculateDiversificationScore()

  return (
    <Card className="p-6">
      <h2 className="text-2xl font-bold mb-4">Advanced Wallet Analysis</h2>
      
      <div className="mb-6">
        <h3 className="text-xl font-semibold mb-2">Portfolio Composition</h3>
        <ResponsiveContainer width="100%" height={300}>
          <PieChart>
            <Pie
              data={tokenDistribution}
              dataKey="value"
              nameKey="name"
              cx="50%"
              cy="50%"
              outerRadius={80}
              fill="#8884d8"
              label
            >
              {tokenDistribution.map((entry, index) => (
                <Cell key={`cell-${index}`} fill={COLORS[index % COLORS.length]} />
              ))}
            </Pie>
            <Tooltip />
            <Legend />
          </PieChart>
        </ResponsiveContainer>
      </div>

      <div className="mb-6">
        <h3 className="text-xl font-semibold mb-2">Diversification Score</h3>
        <p className="text-3xl font-bold">{diversificationScore.toFixed(2)}%</p>
        <p className="text-gray-600">
          {diversificationScore > 70 ? 'Well diversified' : 'Consider diversifying your portfolio'}
        </p>
      </div>

      <div>
        <h3 className="text-xl font-semibold mb-2">Token Performance</h3>
        <ResponsiveContainer width="100%" height={300}>
          <LineChart data={walletData.tokens}>
            <CartesianGrid strokeDasharray="3 3" />
            <XAxis dataKey="symbol" />
            <YAxis />
            <Tooltip />
            <Legend />
            <Line type="monotone" dataKey="value" stroke="#8884d8" />
          </LineChart>
        </ResponsiveContainer>
      </div>
    </Card>
  )
}

