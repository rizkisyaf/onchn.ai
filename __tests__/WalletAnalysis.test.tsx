import { render, screen, waitFor } from '@testing-library/react'
import { WalletAnalysis } from '@/components/wallet-analysis'
import { getWalletData } from '@/lib/api-client'
import { act } from '@testing-library/react'

jest.mock('@/lib/api-client')

describe('WalletAnalysis', () => {
  const mockWalletAddress = '0x123'
  let rendered: ReturnType<typeof render>

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders wallet analysis component', () => {
    render(<WalletAnalysis address={mockWalletAddress} />)
    expect(screen.getByText(/Risk Score/i)).toBeInTheDocument()
  })

  it('displays loading state initially', () => {
    rendered = render(<WalletAnalysis address={mockWalletAddress} />)
    expect(rendered.container).toMatchSnapshot()
  })

  it('updates with wallet data', async () => {
    const mockData = {
      riskScore: 75,
      transactions: [
        {
          type: 'send',
          amount: '10.5',
          token: 'SOL',
          time: '2 mins ago',
        },
      ],
    }

    ;(getWalletData as jest.Mock).mockResolvedValue(mockData)

    rendered = render(<WalletAnalysis address={mockWalletAddress} />)

    await waitFor(() => {
      expect(screen.getByText('75%')).toBeInTheDocument()
    })
  })
})

