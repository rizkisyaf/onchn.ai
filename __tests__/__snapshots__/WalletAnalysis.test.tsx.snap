// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletAnalysis displays loading state initially 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <div
      style="opacity: 0; transform: translateY(20px);"
    >
      <div
        class="bg-white/5 rounded-xl p-6 backdrop-blur-sm border border-white/10"
      >
        <div
          class="flex items-center justify-between mb-4"
        >
          <div>
            <h3
              class="text-xl font-semibold"
            >
              Risk Score
            </h3>
            <p
              class="text-gray-400"
            >
              Overall wallet safety assessment
            </p>
          </div>
          <div
            class="text-right"
          >
            <p
              class="text-3xl font-bold bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent"
            >
              75
              %
            </p>
            <p
              class="text-sm text-gray-400"
            >
              Safe
            </p>
          </div>
        </div>
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="relative w-full overflow-hidden rounded-full bg-white/5 h-2"
          data-max="100"
          data-state="indeterminate"
          role="progressbar"
        >
          <div
            class="h-full w-full flex-1 bg-gradient-to-r from-primary to-accent transition-all"
            data-max="100"
            data-state="indeterminate"
            style="transform: translateX(-25%);"
          />
        </div>
      </div>
    </div>
    <div
      style="opacity: 0; transform: translateY(20px);"
    >
      <div
        class="space-y-4"
        data-orientation="horizontal"
        dir="ltr"
      >
        <div
          aria-orientation="horizontal"
          class="inline-flex h-10 items-center justify-center rounded-lg p-1 bg-white/5 border-b border-white/10"
          data-orientation="horizontal"
          role="tablist"
          style="outline: none;"
          tabindex="0"
        >
          <button
            aria-controls="radix-:r4:-content-transactions"
            aria-selected="true"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-white/5 data-[state=active]:shadow-sm data-[state=active]:text-primary"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="active"
            id="radix-:r4:-trigger-transactions"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Transactions
          </button>
          <button
            aria-controls="radix-:r4:-content-activity"
            aria-selected="false"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-white/5 data-[state=active]:shadow-sm data-[state=active]:text-primary"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r4:-trigger-activity"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Activity
          </button>
          <button
            aria-controls="radix-:r4:-content-connections"
            aria-selected="false"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-white/5 data-[state=active]:shadow-sm data-[state=active]:text-primary"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r4:-trigger-connections"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Connections
          </button>
        </div>
        <div
          aria-labelledby="radix-:r4:-trigger-transactions"
          class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 space-y-4"
          data-orientation="horizontal"
          data-state="active"
          id="radix-:r4:-content-transactions"
          role="tabpanel"
          style="animation-duration: 0s;"
          tabindex="0"
        >
          <div
            class="bg-white/5 rounded-xl p-4 backdrop-blur-sm border border-white/10 flex items-center justify-between"
          >
            <div
              class="flex items-center gap-4"
            >
              <svg
                class="lucide lucide-arrow-up-down text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21 16-4 4-4-4"
                />
                <path
                  d="M17 20V4"
                />
                <path
                  d="m3 8 4-4 4 4"
                />
                <path
                  d="M7 4v16"
                />
              </svg>
              <div>
                <p
                  class="font-medium"
                >
                  Send
                </p>
                <p
                  class="text-sm text-gray-400"
                >
                  2 mins ago
                </p>
              </div>
            </div>
            <p
              class="font-medium"
            >
              -
              10.5
               
              SOL
            </p>
          </div>
          <div
            class="bg-white/5 rounded-xl p-4 backdrop-blur-sm border border-white/10 flex items-center justify-between"
          >
            <div
              class="flex items-center gap-4"
            >
              <svg
                class="lucide lucide-arrow-up-down text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21 16-4 4-4-4"
                />
                <path
                  d="M17 20V4"
                />
                <path
                  d="m3 8 4-4 4 4"
                />
                <path
                  d="M7 4v16"
                />
              </svg>
              <div>
                <p
                  class="font-medium"
                >
                  Receive
                </p>
                <p
                  class="text-sm text-gray-400"
                >
                  5 mins ago
                </p>
              </div>
            </div>
            <p
              class="font-medium"
            >
              +
              25.0
               
              USDC
            </p>
          </div>
          <div
            class="bg-white/5 rounded-xl p-4 backdrop-blur-sm border border-white/10 flex items-center justify-between"
          >
            <div
              class="flex items-center gap-4"
            >
              <svg
                class="lucide lucide-arrow-up-down text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21 16-4 4-4-4"
                />
                <path
                  d="M17 20V4"
                />
                <path
                  d="m3 8 4-4 4 4"
                />
                <path
                  d="M7 4v16"
                />
              </svg>
              <div>
                <p
                  class="font-medium"
                >
                  Swap
                </p>
                <p
                  class="text-sm text-gray-400"
                >
                  10 mins ago
                </p>
              </div>
            </div>
            <p
              class="font-medium"
            >
              -
              5.2
               
              SOL
            </p>
          </div>
        </div>
        <div
          aria-labelledby="radix-:r4:-trigger-activity"
          class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 space-y-4"
          data-orientation="horizontal"
          data-state="inactive"
          hidden=""
          id="radix-:r4:-content-activity"
          role="tabpanel"
          tabindex="0"
        />
        <div
          aria-labelledby="radix-:r4:-trigger-connections"
          class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 space-y-4"
          data-orientation="horizontal"
          data-state="inactive"
          hidden=""
          id="radix-:r4:-content-connections"
          role="tabpanel"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</div>
`;
